### PR TITLE
Made reload current project ask for confirmation with unsaved changes

### LINKED ISSUE #52287: Reload Current Project doesn't ask for confirmation even with unsaved changes
### Godot version

0ee1179 / 3.x

### System information

W10

### Issue description

When you use Project -> Reload Current Project, it never asks for confirmation, even you have unsaved changes. This leads to potential data loss.

### Steps to reproduce

1. Modify scene (don't save)
2. Reload Current Project

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by vitika9**
Fix #52287
I tested that this (and quit) doesn't work when save each scene on quit setting is turned off, idk if issue is reported or not but I left this problem as is because it should be covered in a separate PR.

### CODE REVIEW

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -2820,29 +2827,35 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
...
 					_discard_changes();
 					break;
 				} else {
 					if (save_each) {
-						_menu_option_confirm(p_option == FILE_QUIT ? FILE_CLOSE_ALL_AND_QUIT : FILE_CLOSE_ALL_AND_RUN_PROJECT_MANAGER, false);
+						if (p_option == RUN_RELOAD_CURRENT_PROJECT) {
+							_menu_option_confirm(FILE_CLOSE_ALL_AND_RUN_RELOAD_CURRENT_PROJECT, false);
+						} else {
+							_menu_option_confirm(p_option == FILE_QUIT ? FILE_CLOSE_ALL_AND_QUIT : FILE_CLOSE_ALL_AND_RUN_PROJECT_MANAGER, false);
+						}
```
Comment (KoBeWi) [inline editor/editor_node.cpp:2844]: The ternary operator could be changed into another if. Right now it's a bit awkward.

Comment (akien-mga) [reply editor/editor_node.cpp:2844]: Even better, a `switch (p_option)`.

Comment (KoBeWi) [reply editor/editor_node.cpp:2844]: I was going to suggest that, but then it's switch inside switch.

Comment (akien-mga) [reply editor/editor_node.cpp:2844]: Ah right.

File: editor/editor_node.h [hunk 2]
```diff
@@ -153,6 +153,7 @@ class EditorNode : public Node {
 		FILE_CLOSE_ALL,
 		FILE_CLOSE_ALL_AND_QUIT,
 		FILE_CLOSE_ALL_AND_RUN_PROJECT_MANAGER,
+		FILE_CLOSE_ALL_AND_RUN_RELOAD_CURRENT_PROJECT,
```
Comment (akien-mga) [inline editor/editor_node.h:156]: This should likely be:
```suggestion
		FILE_CLOSE_ALL_AND_RELOAD_CURRENT_PROJECT,
```

The actions are "quit", "run project manager", and "reload current project".

Comment (Vitika9) [reply editor/editor_node.h:156]: It's was RUN_RELOAD_CURRENT_PROJECT, anyway I will change it.

File: editor/editor_node.cpp (hunk 1 above)
Comment (Vitika9) [reply editor/editor_node.cpp:2844]: > The ternary operator could be changed into another if. Right now it's a bit awkward.
> 
> EDIT: Same below. (that code really needs to be refactored...)

Ok, I will change it.

File: editor/editor_node.h (hunk 2 above)
Comment (akien-mga) [reply editor/editor_node.h]: What I meant is that "run reload" is two verbs and sounds weird. If the action is that the edit is quit and the current project is reloaded, the "run" verb doesn't need to be included.

Comment (Vitika9) [reply editor/editor_node.h]: Got it, does everything seem fine now or do I need to make any changes?

**Review by KoBeWi — APPROVED** [review]

File: editor/editor_node.h [hunk 3]
```diff
@@ -169,7 +170,7 @@ class EditorNode : public Node {
 		RUN_PLAY_CUSTOM_SCENE,
 		RUN_SETTINGS,
 		RUN_USER_DATA_FOLDER,
-		RUN_RELOAD_CURRENT_PROJECT,
+		RELOAD_CURRENT_PROJECT,
```
Comment (akien-mga) [inline editor/editor_node.h:173]: Ah I had missed that `FILE_CLOSE_ALL_AND_RUN_RELOAD_CURRENT_PROJECT` was matching a pre-existing `RUN_RELOAD_CURRENT_PROJECT`. It seems like contributors over time have all prefixed their new actions with `RUN_` without understanding that the original `RUN_` actions are the actions to actually run the project/a scene.

So `RUN_SETTINGS`, `RUN_USER_DATA_FOLDER`, `RUN_PROJECT_MANAGER` are all misnomers. But this is material for a different PR refactoring that code.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Comment by belzecue** [comment]
Minor regression in the 3.5 fix.

3.4.4-stable behavior:

- hit PROJECT > RELOAD CURRENT PROJECT
- pending script changes auto-saved
- pending scenetree changes auto-discarded <--- bug
- Project reloaded WITH PREVIOUSLY ACTIVE SCENE OPEN AND ACTIVE

3.5-RC6 behavior:

- hit PROJECT > RELOAD CURRENT PROJECT
- if pending changes (script or scenetree), "Please confirm" dialog asking to save changes <--- bug fixed, awesome!
- Project reloaded with no active scene, i.e. PREVIOUSLY ACTIVE SCENE DISCARDED <-- regression

For me, this is a small hit to usability, because it was nice previously to just hit "project reload" (after quickly hitting ctrl-shift-alt-S to force global save, due to the bug) when you need to clear any lingering editor issues and then land right back in the same scene after reload to continue on.  But now we have to go find the scene/script we were working on and reopen it.

No biggie, but just noting the change in behavior (which might be due to the way the fix had to be implemented).

**Comment by belzecue** [comment]
Minor change (not really a regression) in behavior after the 3.5 fix:

3.4.4-stable behavior:

- Make a change to the scenetree (don't manually save the change)
- hit PROJECT > RELOAD CURRENT PROJECT
- pending scenetree changes auto-discarded <--- bug
- Project reloads WITH PREVIOUSLY ACTIVE SCENE/TAB OPEN AND ACTIVE

3.5-RC6 behavior:

- Make a change to the scenetree (don't manually save the change)
- hit PROJECT > RELOAD CURRENT PROJECT
- "Please confirm" dialog asking to save changes <--- bug fixed, awesome!
- Hit "Save and Reload"
- Project reloaded with no active scene, i.e. PREVIOUSLY ACTIVE SCENE NOT REOPENED

For me, this is a small hit to usability, because it was nice previously to just hit "project reload" (after quickly hitting ctrl-shift-alt-S to force global save, due to the bug) when you need to clear any lingering editor issues and then land right back in the same scene after reload to continue on.  But now we have to go find the scene we were working on and reopen it.

Not a big deal, but just noting the change in behavior.  You can observe how the "save and reload" step now closes the active scene, leaving just the "empty" tab before reloading the project, so I guess that new step is why it doesn't reopen the previously active scene on reload.

Note that in both 3.4.4 and 3.5-RC6, if you ONLY make an unsaved change to a script and hit RELOAD CURRENT PROJECT then it autosaves WITHOUT showing the "do you want to save" dialog, and in both cases the project will be reloaded with the previously active script tab open and active again.  This post-fix change (no longer reopens the active scene) only exhibits when there are unsaved changes to the scenetree.

**Comment by KoBeWi** [comment]
Sounds similar to #33326
Not sure why it would happen only in 3.x though.

**Pull request merged**